src/Core/SettingsChangesHistory.cpp

```diff
@@ -77,6 +77,7 @@ static std::initializer_list<std::pair<ClickHouseVersion, SettingsChangesHistory
             {"backup_restore_keeper_max_retries_while_handling_error", 0, 20, "New setting."},
             {"backup_restore_finish_timeout_after_error_sec", 0, 180, "New setting."},
             {"parallel_replicas_local_plan", false, true, "Use local plan for local replica in a query with parallel replicas"},
+            {"merge_tree_use_v1_object_and_dynamic_serialization", true, false, "Add new serialization V2 version for JSON and Dynamic types"}
```

Comment:
Doesn't it make more sense to make it a server setting? I'm not sure we want this to be available for changing during queries, for example.

Reply:
Does `compatibility` setting allows to change server settings? If not, then better to have it here. Also from the code, this setting takes affect only when enabled for global context (probably need to make a note that changing this settings requires server restart)